libfwupd/fwupd-bios-attr.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2022 Mario Limonciello <mario.limonciello@amd.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+#include <glib-object.h>
+
+#include "fwupd-enums.h"
+
+G_BEGIN_DECLS
+
+#define FWUPD_TYPE_BIOS_ATTR (fwupd_bios_attr_get_type())
+G_DECLARE_DERIVABLE_TYPE(FwupdBiosAttr, fwupd_bios_attr, FWUPD, BIOS_ATTR, GObject)
+
+struct _FwupdBiosAttrClass {
+	GObjectClass parent_class;
+	/*< private >*/
+	void (*_fwupd_reserved1)(void);
+	void (*_fwupd_reserved2)(void);
+	void (*_fwupd_reserved3)(void);
+	void (*_fwupd_reserved4)(void);
+	void (*_fwupd_reserved5)(void);
+	void (*_fwupd_reserved6)(void);
+	void (*_fwupd_reserved7)(void);
+};
+
+/* special attributes */
+#define FWUPD_BIOS_ATTR_PENDING_REBOOT "pending_reboot"
+#define FWUPD_BIOS_ATTR_RESET_BIOS     "reset_bios"
+#define FWUPD_BIOS_ATTR_DEBUG_CMD      "debug_cmd"
+
+/**
+ * FwupdBiosAttrKind:
+ * @FWUPD_BIOS_ATTR_KIND_UNKNOWN:		BIOS attribute type is unknown
+ * @FWUPD_BIOS_ATTR_KIND_READ_ONLY:		BIOS attribute that can't be written
+ * @FWUPD_BIOS_ATTR_KIND_ENUMERATION:		BIOS attribute that has enumerated possible values
+ * @FWUPD_BIOS_ATTR_KIND_INTEGER:		BIOS attribute that is an integer
+ * @FWUPD_BIOS_ATTR_KIND_STRING:		BIOS attribute that accepts a string
```

Comment:
is this how they're enumerated in the kernel? I just feel that that are a bit more like bitfields, i.e. can I have a read-only enumeration?

Reply:
Everything except read only is enumerated this way by the kernel.  Read only is a special concept added to fwupd for `pending_reboot`.  This is technically read only by the kernel too but doesn't have a type.